packaging/flavorgen/flavors/generators.go

```diff
@@ -246,10 +246,9 @@ func defaultKubeadmInitSpec(files []bootstrapv1.File) bootstrapv1.KubeadmConfigS
 			},
 			ControllerManager: defaultControlPlaneComponent(),
 		},
-		Users:                    defaultUsers(),
-		PreKubeadmCommands:       defaultPreKubeadmCommands(),
-		UseExperimentalRetryJoin: true,
```

Comment:
why did we drop `UseExperimentalRetryJoin` it's still needed in slow/flaky envs

Reply:
This has been deprecated in kubeadmconfig types. Checkout https://github.com/kubernetes-sigs/cluster-api/blob/8b4214d72762394144b83dd6d14986ff7e274095/bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go#L100-L101